clang/lib/Driver/Driver.cpp

```diff
@@ -1326,6 +1326,20 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
       }
     }
   }
+  // -fsycl-fp64-conv-emu is valid only for AOT compilation with an Intel GPU
+  // target. For other scenarios, we emit a warning message.
+  if (C.getInputArgs().hasArg(options::OPT_fsycl_fp64_conv_emu)) {
+    bool HasIntelGPUAOTTarget = false;
+    for (auto &TT : UniqueSYCLTriplesVec) {
+      if (TT.isSPIRAOT() && TT.getSubArch() == llvm::Triple::SPIRSubArch_gen) {
+        HasIntelGPUAOTTarget = true;
+        break;
+      }
+    }
+    if (!HasIntelGPUAOTTarget)
+      Diag(diag::warn_invalid_fp64_emu_use);
```

Comment:
OK by me - just a NIT, use of the bool isn't needed and the diag can be emitted in the loop before the break.

Reply:
There is a negation here though...

Reply:
Oh, my bad - must have read through that a bit too fast!